client/src/partials/cotisation/create/create_cotisation.js

```diff
@@ -98,29 +95,30 @@ angular.module('bhima.controllers')
     };
 
     $scope.save.new = function () {
-      $scope.session.new.is_employee = ($scope.session.new.is_employee)? 1 : 0;
-      $scope.session.new.is_percent = ($scope.session.new.is_percent)? 1 : 0;
       var record = connect.clean(session.new);
-      record.six_account_id = session.new.six_account_id;
-      
-      if(record.abbr){
-        if(record.abbr.length <= 4){
-          connect.basicPut('cotisation', [record])
-          .then(function (res) {
 
-            validate.refresh(dependencies)
-            .then(function (models) {
-              angular.extend($scope, models);
-            });
-            session.action = '';
-            session.new = {};
-          });
-        } else if (record.abbr.length > 4){
-          messenger.danger($translate.instant('COTISATIONS.NOT_SUP4'));
-        }
-      }  else {
-        messenger.danger($translate.instant('COTISATIONS.NOT_EMPTY'));
+      // fail if no abbreviation
+      if (!record.abbr) {
+        return messenger.danger($translate.instant('COTISATIONS.NOT_EMPTY'));
       }
+
+      // require small abbreviations
+      if (record.abbr.length > 4) {
+        return messenger.danger($translate.instant('COTISATIONS.NOT_SUP4'));
+      }
+      
+      // all checks pass, we are free to post to the server!
+      connect.basicPut('cotisation', [record])
```

Comment:
I didn't want to change the code too much; but you are right, post makes sense.  I'm make the changes.